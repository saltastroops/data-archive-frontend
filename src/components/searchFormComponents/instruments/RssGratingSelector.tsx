import * as React from "react";
import MultiSelectField, {
  AllOption
} from "../../basicComponents/MultiSelectField";

interface IRssGratingSelectorProps {
  rssGratings?: string[];
  onSelect: (value: any) => void;
}

// Rss grating
export const RSS_GRATINGS = [
  "Open",
  "pg0300",
  "pg0900",
  "pg1300",
  "pg1800",
  "pg2300",
  "pg3000"
];

/**
 * A form for selecting RssGrating-related search parameters.
 */
const RssGratingSelector = (props: IRssGratingSelectorProps) => {
  const { onSelect, rssGratings } = props;

  // Function for handling change events
  const select = (e: React.SyntheticEvent<HTMLSelectElement>) => {
    const values = e.currentTarget.selectedOptions;
    onSelect({
      rssGratings: Array.from(values).map((t: any) => {
        return t.text;
      })
    });
  };

  return (
    <div>
      <p>RSS grating</p>
      <MultiSelectField
        name={"rssGratings"}
        onChange={select}
        value={rssGratings || ["All"]}
      >
        <AllOption />
        {RSS_GRATINGS.sort().map(i => (
          <option key={i} value={i}>
            {i}
          </option>
        ))}
      </MultiSelectField>
    </div>
  );
};

export default RssGratingSelector;
