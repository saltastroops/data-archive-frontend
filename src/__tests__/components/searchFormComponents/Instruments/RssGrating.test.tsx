import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import RssGratingSelector from "../../../../components/searchFormComponents/instruments/RssGratingSelector";

const onSelect = jest.fn() as any;
describe("RssGrating", () => {
  it("should be defined", () => {
    expect(
      mount(<RssGratingSelector rssGratings={[]} onSelect={onSelect} />)
    ).toBeDefined();
  });

  it("should render correctly", () => {
    // Use mount instead of shallow for better snapshots
    expect(
      toJson(mount(<RssGratingSelector rssGratings={[]} onSelect={onSelect} />))
    ).toMatchSnapshot();
  });
});
