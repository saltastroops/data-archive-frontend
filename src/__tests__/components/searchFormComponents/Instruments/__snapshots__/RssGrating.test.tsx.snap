// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RssGrating should render correctly 1`] = `
.c0 {
  width: 95%;
}

<RssGratingSelector
  onSelect={[MockFunction]}
  rssGratings={Array []}
>
  <div>
    <p>
      RSS grating
    </p>
    <MultiSelectField
      name="rssGratings"
      onChange={[Function]}
      value={Array []}
    >
      <styled.div
        className="control select is-multiple "
      >
        <StyledComponent
          className="control select is-multiple "
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "c0",
                "rules": Array [
                  "
  width: 95%;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="control select is-multiple  c0"
          >
            <select
              multiple={true}
              name="rssGratings"
              onChange={[Function]}
              size={6}
              style={
                Object {
                  "width": "216px",
                }
              }
              value={Array []}
            >
              <AllOption>
                <option
                  value="All"
                >
                  All
                </option>
              </AllOption>
              <option
                key="Open"
                value="Open"
              >
                Open
              </option>
              <option
                key="PG0300"
                value="PG0300"
              >
                PG0300
              </option>
              <option
                key="PG0900"
                value="PG0900"
              >
                PG0900
              </option>
              <option
                key="PG1300"
                value="PG1300"
              >
                PG1300
              </option>
              <option
                key="PG1800"
                value="PG1800"
              >
                PG1800
              </option>
              <option
                key="PG2300"
                value="PG2300"
              >
                PG2300
              </option>
              <option
                key="PG3000"
                value="PG3000"
              >
                PG3000
              </option>
            </select>
          </div>
        </StyledComponent>
      </styled.div>
    </MultiSelectField>
  </div>
</RssGratingSelector>
`;
